compiler/optimizer/Dominators.hpp

```diff
@@ -55,30 +54,26 @@ class TR_Dominators
    int             dominates(TR::Block *block, TR::Block *other);
 
    TR::Compilation * comp()         { return _compilation; }
-   TR_Memory *      trMemory()      { return comp()->trMemory(); }
-   TR_StackMemory   trStackMemory() { return trMemory(); }
    bool trace() { return _trace; }
 
-   TR::deque<int32_t> _dfNumbers;
-
    protected:
 
-   typedef TR::SparseBitVector SparseBitVector;
+   typedef TR_BitVector BitVector;
 
    private:
 
    struct BBInfo
       {
-      BBInfo(TR::Allocator allocator) :
-         _bucket(allocator), _block(NULL), _parent(-1), _idom(-1), _ancestor(-1), _label(-1), _child(-1), _sdno(-1), _size(0)
+      explicit BBInfo(TR::Region &region) :
+         _bucket(region), _block(NULL), _parent(-1), _idom(-1), _ancestor(-1), _label(-1), _child(-1), _sdno(-1), _size(0)
          {}
       TR::Block     *_block;     // The block whose info this is
       int32_t       _parent;    // The parent in the depth-first spanning tree
       int32_t       _idom;      // The immediate dominator for this block
       int32_t       _ancestor;  // The ancestor in the forest
       int32_t       _label;     // The node in the ancestor chain with minimal
                                 // semidominator number
-      SparseBitVector _bucket;    // The blocks whose semidominator is this node
+      BitVector _bucket;    // The blocks whose semidominator is this node
```

Comment:
So the issue here is that the BitVector that was being used was a CS2 sparse bit vector. The sparseness is good, but it allocates its memory from the heap and it is not recycled effectively due to heap fragmentation. The TR_BitVector is semi-sparse (compared to the CS2 dense BitVector). but the major advantage here is that the TR_BitVector is compatible with region allocators and so the memory can be fully cleaned up. I changed the bit vector for that reason - all the data structures from the analysis now are allocated in a region and that region will be fully cleaned up leading to lower heap memory usage over the course of a compile rather than the slow leak seen with the CS2 allocator for the SparseBitVector.